src/usr/share/netcheck/netcheck.py

```diff
@@ -776,3 +804,59 @@ def _calculate_next_log_time(self, loop_time):
         Returns the datetime that the logging should occur.
         """
         return loop_time + datetime.timedelta(seconds=self.config['periodic_status_delay'])
+
+    def _check_for_gateway_change(self):
+        """Checks the current state of the default gateway and issues a broadcast if it is
+        different from the prior state.
+        """
+
+        new_default_gateway_state = self._get_default_gateway_state()
+
+        if new_default_gateway_state \
+            and new_default_gateway_state != self.default_gateway_state:
+
+            self.logger.info(
+                'The default gateway has changed to %s via %s on interface %s.',
+                new_default_gateway_state['address'],
+                new_default_gateway_state['connection_id'],
+                new_default_gateway_state['interface'])
+
+            self.broadcaster.issue()
+
+        self.default_gateway_state = new_default_gateway_state
+
+    def _get_default_gateway_state(self):
+        """Retrieves information about the current primary default gateway.
+
+        Returns a dictionary containing a gateway IP address, interface name, and associated
```

Comment:
I commented on this already but you might have missed it. I'd replace 'a' with 'the'.